packages/react/runtime/src/backgroundSnapshot.ts

```diff
@@ -254,9 +254,9 @@ export class BackgroundSnapshotInstance {
             if (!newSpreadValue) {
               continue;
             }
-            if ((newSpreadValue as any)._wkltId) {
+            if (!__DISABLE_MTS_AND_GESTURE__ && (newSpreadValue as any)._wkltId) {
```

Comment:
Instead of using a macro, can we use something like a pluginSystem?

E.g.:

```js
// backgroundSnapshot.ts
for (let key in newSpread) {
  const newSpreadValue = newSpread[key]
  newSpread[key] = spreadPlugins.call(newSpreadValue)
}
```

And we only register MTS & gesture related plugins when using MTS/gesture related APIs

```js
export const runOnBackground = /*#__PURE__*/ (() => {
  spreadPlugins.tap('mtx', onPostWorkletCtx)

  return function runOnBackground() { /* impl */ }
})()
```

/cc @Yradex